packages/front-end/components/importing/ImportFromLaunchDarkly/useImportFromLaunchDarkly.ts

```diff
@@ -0,0 +1,700 @@
+                  status: "fail",
+                  error: response.message || "unknown error",
+                };
+              }
+
+              return {
+                status: "success",
+                data: response.feature,
+              };
+            } catch (e) {
+              return {
+                status: "fail",
+                error: e.message || "unknown error",
+              };
+            }
+          },
+        });
+      };
+
+      const createAll = async () => {
+        await createEnvironments();
+        await createFeatures();
+
+        setStatus("completed");
+      };
+
+      createAll();
+    },
+    [
+      apiToken,
+      state.projectsReady,
+      state.gbProjectsCreated,
+      state.gbEnvironments,
+      apiCall,
+      status,
+    ]
+  );
+
+  useEffect(
+    /**
+     * This runs once the projects are downloaded from LD and created in GrowthBook.
+     * Projects are a hard dependency of remaining resources in LD and
+     * without them we cannot proceed.
+     */
+    function fetchLDResourcesForProjects() {
+      if (!state.gbProjects.length) {
+        return;
+      }
+      if (!apiToken) {
+        console.warn("cannot fetch LD resources without API token");
+        return;
+      }
+
+      const perform = async () => {
+        const createProjectTasks: QueueTask<
+          Pick<ProjectInterface, "name" | "description">
+        >[] = state.gbProjects.map((p) => ({
+          id: p.name,
+          data: p,
+        }));
+
+        await enqueueTasks<
+          Pick<ProjectInterface, "name" | "description">,
+          ProjectInterface
+        >(createProjectTasks, {
+          onProgress(id, result): void {
+            switch (result.status) {
+              case "success":
+                dispatch({
+                  type: "add-gb-imported-project",
+                  data: result.data,
+                });
+                dispatch({
+                  type: "add-import-project-result",
+                  data: {
+                    status: "completed",
+                    message: `Imported project ${id} as ${result.data.name} (${result.data.id})`,
+                  },
+                });
+                break;
+
+              case "retry":
+              case "fail":
+                dispatch({
+                  type: "add-import-project-result",
+                  data: {
+                    status: "failed",
+                    message: `Failed to import project ${id} with error: ${result.error}`,
+                  },
+                });
+                break;
+            }
+          },
+          async perform(data): Promise<TaskResult<ProjectInterface>> {
+            try {
+              const response = await apiCall<{
+                project?: ProjectInterface;
+                message?: string;
+              }>("/projects", {
+                method: "POST",
+                body: JSON.stringify(data),
+              });
```

Comment:
If you run the import multiple times, will this create duplicate projects?  I think for environments and features, it will not create duplicates, but this one I'm not sure.

Reply:
Yes, duplicate projects will be created. There's no backend-enforced uniqueness constraint on the project name so this succeeds. If this is a problem, I can check with the projects returned from `useDefinitions()` and do this validation client-side.

Reply:
I've updated this. I'm checking the projects from `useDefinitions()` before creating.